### DISCUSSION TITLE
Strapi updated content SSR Nextjs cache

### DISCUSSION THREAD

**Discussion by field123** [urql-graphql/urql#1931] [Q&A] [answered]
With the following scenario I can't seem to find a solution.

urql setup
```
const isServerSide = typeof window === 'undefined';

const ssrCache = ssrExchange({
  isClient: !isServerSide,
});

const client = createClient({
  url: getStrapiGraphQLEndpoint(),
  exchanges: [
    dedupExchange,
    cacheExchange,
    ssrCache,
    fetchExchange,
  ],
});
```

1. query for page data from Strapi CMS back end on a nextjs front end
```
const JoinMailingListQuery = `
query {
  joinMailingList{
    id
    title
  }
}
`;

export async function getServerSideProps() {
  await client.query(JoinMailingListQuery).toPromise();
  return {
    props: {
      urqlState: ssrCache.extractData(),
    },
  };
}
```
2. Update data outside in the external Strapi admin ui e.g. change title
3. Invalidate cache after x amount of time
4. new version of the page data is passed to the component and rendered

What is the generally suggested way to setup caching in urql to enable this situations? I can't seem to understand how to invalidate a local graphql query cache entry after a fixed amount of time.

Appreciate any guidance you can offer.

Edited:

I've discovered the requestPolicy but still can't find the behaviour I'm looking for, a time to live or max age type experation would be idle.

```
const client = createClient({
  url: getStrapiGraphQLEndpoint(),
  requestPolicy: ['cache-first' | 'cache-only' | 'network-only' | 'cache-and-network],
  exchanges: [
    dedupExchange,
    cacheExchange,
    ssrCache,
    fetchExchange,
  ],
});
```

**Comment by JoviDeCroock** [comment, marked as answer]
I think you're looking for this exchange https://github.com/FormidableLabs/urql/blob/main/exchanges/request-policy/README.md

**Reply by field123** [reply]
That looks like it may be it will give a go and report back.
